### PR TITLE
First pull request

### PR DESCRIPTION
**Pull request by demar01**
Hi Jesse

These are the changes I made on PeCorA. I changed the README.md quite a bit; we can decide later what we want to say/show there.

### CODE REVIEW

**Review by jessegmeyerlab — CHANGES_REQUESTED** [review]
Readmes:
	

- 

- [ ] change section 2 header to "install":

- [ ] add windows install instructions 
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;within R, development version
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;install.packages("devtools")
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;devtools::install_github("DeveloperName/PackageName") 	

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;within R, CRAN?
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;install.packages("PeCorA")

- [ ] update install location to jessegmeyerlab repo

- [ ] maybe under header 3, add subsection on how to use their own data, eg what columns/information are required

- [ ] Functions needed?

- &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;convert maxquant peptides.txt to pecora format 
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;would need user to input peptides.txt and unique group name from the LFQ col names
- &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;wrapper function to run via command line


loading into R saw this, is this a problem?:
> library(PeCorA)
Registered S3 methods overwritten by 'lme4':
  method                          from
  cooks.distance.influence.merMod car 
  influence.merMod                car 
  dfbeta.influence.merMod         car 
  dfbetas.influence.merMod        car

Testing vingette got error:
> scaled_peptides<-PeCorA_preprocessing(t, 8, 100, "cntrl")
Error in scale_by(ms1log2 ~ Condition * BioReplicate, t) : 
  could not find function "scale_by"
In addition: Warning message:
In which(is.na(as.numeric(t[, area_column_name])) == TRUE) :
  NAs introduced by coercion

**Comment by demar01** [comment]
> Readmes:
> 
> * [ ]  change section 2 header to "install":
Changed 
> * [ ]  add windows install instructions
> * within R, development version
>         install.packages("devtools")
>         devtools::install_github("jessegmeyerlab/PeCorA")
Changed
> * within R, CRAN?
>         install.packages("PeCorA")
Added that but still not submited to CRAN
> * [ ]  update install location to jessegmeyerlab repo
Changed
> * [ ]  maybe under header 3, add subsection on how to use their own data, eg what columns/information are required
Changed
> * [ ]  Functions needed?
> * convert maxquant peptides.txt to pecora format
>               would need user to input peptides.txt and unique group name from the LFQ col names
> * wrapper function to run via command line
Need to add those function to package
> loading into R saw this, is this a problem?:
> 
> > library(PeCorA)
> > Registered S3 methods overwritten by 'lme4':
> > method                          from
> > cooks.distance.influence.merMod car
> > influence.merMod                car
> > dfbeta.influence.merMod         car
> > dfbetas.influence.merMod        car

PeCorA package is loaded quietly in the Vignette now, so that message does not appear

> Testing vignette got error, does function need to be standardize::scale_by?:
> 
> > scaled_peptides<-PeCorA_preprocessing(t, 8, 100, "cntrl")
> > Error in scale_by(ms1log2 ~ Condition * BioReplicate, t) :
> > could not find function "scale_by"

library(standardize) is also added in the vignette now, which removes the error. standardize::scale_by is imported in PeCoRa package, but it doesn't seem to be imported correctly for now. 

> > In addition: Warning message:
> > In which(is.na(as.numeric(t[, area_column_name])) == TRUE) :
> > NAs introduced by coercion

I need to suppress that warning in PeCorA_preprocessing function.

File: R/PeCorA_preprocessing.R [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  if(length(which(is.na(as.numeric(t[,area_column_name]))==TRUE))>0){
+    t <- t[-suppressWarnings(which(is.na(as.numeric(t[,area_column_name]))==TRUE)),]
+  }
+  t <- t[-which(as.numeric(t[,area_column_name])<=threshold_to_filter),]
+  t <- t[t$modpep_z %in% names(which(table(t$modpep_z)==n_reps_total)),]
+
+
+  ## scale the data and then plot
+  t$ms1log2<-log2(as.numeric(t[,area_column_name]))
+  t$ms1scaled <- scale_by(ms1log2 ~ Condition*BioReplicate, t)
```
Comment (jessegmeyerlab) [inline R/PeCorA_preprocessing.R:37]: does this need to be changed to?
t$ms1scaled <- standardize::scale_by(ms1log2 ~ Condition*BioReplicate, t)

File: DESCRIPTION [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+RoxygenNote: 7.1.1
+Depends: 
+    R (>= 2.10)
+Imports: 
+    here,
+    magrittr,
+    dplyr,
+    MSstats,
+    rlang,
+    standardize,
```
Comment (jessegmeyerlab) [inline DESCRIPTION:24]: might need to move this to under "Depends"?

**Review by jessegmeyerlab — CHANGES_REQUESTED** [review]
Great, thanks for doing this.  

Can you please update the install instructions in the vignette as well? Sorry that is what I meant when I suggested changing the second section to "install", I like "installation". I can change the main readme myself but less sure how to change the vingette afterwards. 

Can we please fix the dependency as well? I looked and maybe it needs to be under "depends" instead of "imports", or it might need to be called in the function as standardize::scale_by

see line comments related to above

Note it also does not correctly import ggplot

For the command line wrapper, please make it dump all the plots of significantly different peptides as before.

File: R/import_processed_data_for_PeCorA.R [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+               measure.vars=grep(colnames(p), pattern="LFQ"))
+  head(pmelt)
+
+  levels(as.factor(pmelt$variable))
+
+  pmelt$Condition = rep(0, nrow(pmelt))
+  pmelt$Condition[grep(pmelt$variable, pattern="_COVID19")]<-"COVID"
+  head(pmelt)
+  pmelt$Condition[grep(pmelt$variable, pattern="NON.COVID19")]<-"NONCO"
+  pmelt$Condition[grep(pmelt$variable, pattern="control")]<-"CNTRL"
```
Comment (jessegmeyerlab) [inline R/import_processed_data_for_PeCorA.R:42]: Need to update this so the function can account for user-specific groups in their maxquant data.

**Review by jessegmeyerlab — CHANGES_REQUESTED** [review]
in addition to the other issues we discussed note the maxquant re-formatting function should be able to look for unique string in the LFQ columns specified by the user to define the quantification groups

File: R/PeCorA_preprocessing.R (hunk 1 above)
Comment (demar01) [reply R/PeCorA_preprocessing.R]: Changed

File: DESCRIPTION (hunk 2 above)
Comment (demar01) [reply DESCRIPTION]: Moved to Depends

File: R/import_processed_data_for_PeCorA.R (hunk 3 above)
Comment (demar01) [reply R/import_processed_data_for_PeCorA.R:42]: Changed in function `import_processed_data_for_PeCorA_LFQ`

i.e.
t2 <- import_processed_data_for_PeCorA_LFQ( LFQfile = "peptides.txt",
condition1="CONTROL",
condition2="_COVID",
condition3="NON.COVID")

**Comment by demar01** [comment]
> Great, thanks for doing this.
> 
> Can you please update the install instructions in the vignette as well? Sorry that is what I meant when I suggested changing the second section to "install", I like "installation". I can change the main readme myself but less sure how to change the vingette afterwards.

Vignette updated. See PeCorA_vignette.pdf

> Can we please fix the dependency as well? I looked and maybe it needs to be under "depends" instead of "imports", or it might need to be called in the function as standardize::scale_by

Changed now

> see line comments related to above
> 
> Note it also does not correctly import ggplot

It does import ggplot correctly now

> For the command line wrapper, please make it dump all the plots of significantly different peptides as before.

**Comment by jessegmeyerlab** [comment]
Thanks for updating this. I just tried to test it and now it wont install at all. It might be some problem with the package dependencies "digest", "reshape", and/or "MSstats". 

`> devtools::install_github("demar01/pecora-1")
Downloading GitHub repo demar01/pecora-1@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                               
2: CRAN packages only                
3: None                              
4: digest   (0.6.25 -> 0.6.26) [CRAN]
5: quantreg (5.73   -> 5.74  ) [CRAN]

Enter one or more numbers, or an empty line to skip updates:3
√  checking for file 'C:\Users\jmeyer\AppData\Local\Temp\Rtmp0i44df\remotes6f5411803d1e\demar01-PeCorA-1-136091f/DESCRIPTION' ...
-  preparing 'PeCorA': (731ms)
√  checking DESCRIPTION meta-information ... 
-  checking for LF line-endings in source and make files and shell scripts
-  checking for empty or unneeded directories
     NB: this package now depends on R (>= 3.5.0)
     WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/Covid_peptides.rda'  'PeCorA/data/PeCorA_noZ.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.dda.iprg.pg.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.microglia.sk.rda'
-  building 'PeCorA_0.0.0.9000.tar.gz'
   
Installing package into ‘C:/Users/jmeyer/Documents/R/win-library/4.0’
(as ‘lib’ is unspecified)
* installing *source* package 'PeCorA' ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
Error: (converted from warning) package 'reshape' was built under R version 4.0.3
Execution halted
ERROR: lazy loading failed for package 'PeCorA'
* removing 'C:/Users/jmeyer/Documents/R/win-library/4.0/PeCorA'
* restoring previous 'C:/Users/jmeyer/Documents/R/win-library/4.0/PeCorA'
Error: Failed to install 'PeCorA' from GitHub:
  (converted from warning) installation of package ‘C:/Users/jmeyer/AppData/Local/Temp/Rtmp0i44df/file6f5473eb67f7/PeCorA_0.0.0.9000.tar.gz’ had non-zero exit status`

**Comment by jessegmeyerlab** [comment]
Tried on my tablet to check if it was install specific and got a different error with MSstats?

`> devtools::install_github("demar01/pecora-1")
Downloading GitHub repo demar01/pecora-1@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                               
2: CRAN packages only                
3: None                              
4: digest   (0.6.25 -> 0.6.26) [CRAN]
5: rlang    (0.4.6  -> 0.4.8 ) [CRAN]
6: quantreg (5.73   -> 5.74  ) [CRAN]

Enter one or more numbers, or an empty line to skip updates:3
√  checking for file 'C:\Users\jesse\AppData\Local\Temp\Rtmpa80Mc0\remotese081b222ca\demar01-PeCorA-1-136091f/DESCRIPTION' (523ms)
-  preparing 'PeCorA': (2.6s)
√  checking DESCRIPTION meta-information ... 
-  checking for LF line-endings in source and make files and shell scripts
-  checking for empty or unneeded directories
     NB: this package now depends on R (>= 3.5.0)
     WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/Covid_peptides.rda'  'PeCorA/data/PeCorA_noZ.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.dda.iprg.pg.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.microglia.sk.rda'
-  building 'PeCorA_0.0.0.9000.tar.gz'
   
Installing package into ‘C:/Users/jesse/Documents/R/win-library/4.0’
(as ‘lib’ is unspecified)
ERROR: dependency 'MSstats' is not available for package 'PeCorA'
* removing 'C:/Users/jesse/Documents/R/win-library/4.0/PeCorA'
Error: Failed to install 'PeCorA' from GitHub:
  (converted from warning) installation of package ‘C:/Users/jesse/AppData/Local/Temp/Rtmpa80Mc0/filee087d506c6e/PeCorA_0.0.0.9000.tar.gz’ had non-zero exit status`

**Comment by demar01** [comment]
> Tried on my tablet to check if it was install specific and got a different error with MSstats?
> 
> `> devtools::install_github("demar01/pecora-1")
> Downloading GitHub repo demar01/pecora-1@HEAD
> These packages have more recent versions available.
> It is recommended to update all of them.
> Which would you like to update?
> 
> 1: All
> 2: CRAN packages only
> 3: None
> 4: digest (0.6.25 -> 0.6.26) [CRAN]
> 5: rlang (0.4.6 -> 0.4.8 ) [CRAN]
> 6: quantreg (5.73 -> 5.74 ) [CRAN]
> 
> Enter one or more numbers, or an empty line to skip updates:3
> √ checking for file 'C:\Users\jesse\AppData\Local\Temp\Rtmpa80Mc0\remotese081b222ca\demar01-PeCorA-1-136091f/DESCRIPTION' (523ms)
> 
> * preparing 'PeCorA': (2.6s)
>   √  checking DESCRIPTION meta-information ...
> * checking for LF line-endings in source and make files and shell scripts
> * checking for empty or unneeded directories
>   NB: this package now depends on R (>= 3.5.0)
>   WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/Covid_peptides.rda'  'PeCorA/data/PeCorA_noZ.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.dda.iprg.pg.rda'  WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'PeCorA/data/input.microglia.sk.rda'
> * building 'PeCorA_0.0.0.9000.tar.gz'
> 
> Installing package into ‘C:/Users/jesse/Documents/R/win-library/4.0’
> (as ‘lib’ is unspecified)
> ERROR: dependency 'MSstats' is not available for package 'PeCorA'
>
> * removing 'C:/Users/jesse/Documents/R/win-library/4.0/PeCorA'
>   Error: Failed to install 'PeCorA' from GitHub:
>   (converted from warning) installation of package ‘C:/Users/jesse/AppData/Local/Temp/Rtmpa80Mc0/filee087d506c6e/PeCorA_0.0.0.9000.tar.gz’ had non-zero exit status`

I moved MSstats to dependency now. If I download it and run R CMD INSTALL PeCorA_0.0.0.9000.tar.gz it works for me. Does it for you? What if you update all the packages ( option 1)?

I made a wrapper to run PeCorA on the command line and updated the instructions in the vignette. It works on my computer, can you check on yours (windows?)

**Comment by jessegmeyerlab** [comment]
Updating the packages didn't help, it seems related to the change to add those packages as "depends". I cloned your repo and will try to figure it out for windows and let you know. 

I moved those back out of depends to dissect the problem and now its complaining that its missing the function "import_processed_data_for_PeCorA_LFQ" 
from the R folder. 

Can you please check that you pushed that to the master? maybe the name is wrong

**Comment by jessegmeyerlab** [comment]
Also not seeing the wrapper function - can you check that you pushed that to your fork?

**Review by demar01 — COMMENTED** [review]
@jessegmeyerlab this is the wrapper function

**Comment by demar01** [comment]
> Updating the packages didn't help, it seems related to the change to add those packages as "depends". I cloned your repo and will try to figure it out for windows and let you know.
> 
> I moved those back out of depends to dissect the problem and now its complaining that its missing the function "import_processed_data_for_PeCorA_LFQ"
> from the R folder.
> 
> Can you please check that you pushed that to the master? maybe the name is wrong
I am  always posting to your master branch from mine: 
image

> `** building package indices
> ** testing if installed package can be loaded from temporary location
> *** arch - i386
> Error: package or namespace load failed for 'PeCorA' in namespaceExport(ns, exports):
> undefined exports: import_processed_data_for_PeCorA_LFQ
> Error: loading failed
> Execution halted
> *** arch - x64
> Error: package or namespace load failed for 'PeCorA' in namespaceExport(ns, exports):
> undefined exports: import_processed_data_for_PeCorA_LFQ
> Error: loading failed
> Execution halted
> ERROR: loading failed for 'i386', 'x64'
> 
> * removing 'C:/Users/jmeyer/Documents/R/win-library/4.0/PeCorA'
>   Warning message:
>   In i.p(...) :
>   installation of package ‘C:/Users/jmeyer/AppData/Local/Temp/RtmpABcgkR/file4bfc1b3d72e5/PeCorA_0.0.0.9000.tar.gz’ had non-zero exit status`

**Comment by jessegmeyerlab** [comment]
> @jessegmeyerlab this is the wrapper function

I see now thanks. I was confused because I didn't see it in the "R" folder. Is that OK to be in the directory?

**Comment by jessegmeyerlab** [comment]
there is no "import_processed_data_for_PeCorA_LFQ" in the folder "R", but it is in the NAMESPACE. I think that is why it is complaining?

**Comment by demar01** [comment]
Lemme check

Maria Dermit
Postdoctoral Research Assistant | Centre for Molecular Oncology

Barts Cancer Institute - a Cancer Research UK Centre of Excellence
Queen Mary, University of London
John Vane Science Centre, Charterhouse Square, London EC1M 6BQ
Email: maria.dermit@qmul.ac.uk
________________________________

**Comment by demar01** [comment]
> there is no "import_processed_data_for_PeCorA_LFQ" in the folder "R", but it is in the NAMESPACE. I think that is why it is complaining?

Correct, it should be there now. Is is working?

**Comment by jessegmeyerlab** [comment]
Thanks! will check again now

**Comment by jessegmeyerlab** [comment]
OK it worked on my computer that already had MSstats installed by devtools::install_github(), but failed on dependency MSstats not available when I didn't have it installed on my tablet. 

MSstats isn't on cran so does it make sense to have as a dependency? Which function is it required for?

**Comment by jessegmeyerlab** [comment]
actually getting a lot of different errors now that MSstats error went away - there is some problem with package versions. I will see if I can figure it out this morning

**Comment by demar01** [comment]
You are right, we dont need MSstats. I remove it from the description. Lemme know if updating packages solve the problem.  Version  of packages uses is on the vignette - which I need to update to remove MSstats.

Maria Dermit
Postdoctoral Research Assistant | Centre for Molecular Oncology

Barts Cancer Institute - a Cancer Research UK Centre of Excellence
Queen Mary, University of London
John Vane Science Centre, Charterhouse Square, London EC1M 6BQ
Email: maria.dermit@qmul.ac.uk
________________________________

**Comment by jessegmeyerlab** [comment]
OK got it to install now. Might have partially been issue with R version so I updated to 4.0.3 and ran it in the base R gui and it installed

going through the vingette and found this error in the control name
image

**Comment by demar01** [comment]
Great, we could write in the documentation:


PeCorA has been tested with R version 3.6.3. The following packages/version are required:

reshape_0.8.8 standardize_0.2.1  ggplot2_3.3.2

what do you mean error in the control name?



Maria Dermit
Postdoctoral Research Assistant | Centre for Molecular Oncology

Barts Cancer Institute - a Cancer Research UK Centre of Excellence
Queen Mary, University of London
John Vane Science Centre, Charterhouse Square, London EC1M 6BQ
Email: maria.dermit@qmul.ac.uk
________________________________

**Comment by jessegmeyerlab** [comment]
Can you see in the image of the vingette? control_name="t<-..." when it should be control_name="cntrl"

> Great, we could write in the documentation: PeCorA has been tested with R version 3.6.3. The following packages/version are required: reshape_0.8.8 standardize_0.2.1 ggplot2_3.3.2 what do you mean error in the control name? Maria Dermit Postdoctoral Research Assistant | Centre for Molecular Oncology Barts Cancer Institute - a Cancer Research UK Centre of Excellence Queen Mary, University of London John Vane Science Centre, Charterhouse Square, London EC1M 6BQ Email: maria.dermit@qmul.ac.uk
> […](#)
> ________________________________ From: jessegmeyerlab <notifications@github.com> Sent: Monday, October 19, 2020 3:29 PM To: jessegmeyerlab/PeCorA <PeCorA@noreply.github.com> Cc: Maria Dermit <maria.dermit@qmul.ac.uk>; Author <author@noreply.github.com> Subject: Re: [jessegmeyerlab/PeCorA] First pull request (#2) OK got it to install now. Might have partially been issue with R version so I updated to 4.0.3 and ran it in the base R gui and it installed going through the vingette and found this error in the control name [image]<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F65188012%2F96464523-c75a6700-11dc-11eb-8068-3789af3f2281.png&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834445860%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=yiZUqW1qHuXIA2SAfk37ysTi3WP6WiLyiA1qtJtVRwA%3D&reserved=0> — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjessegmeyerlab%2FPeCorA%2Fpull%2F2%23issuecomment-712200875&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834455862%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=R8lVmTEzy1o4UUlhF4Cuc0qV2e4yaDKJDF38sMWib0o%3D&reserved=0>, or unsubscribe<https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABGGWHPORNMDIAEZT346ABLSLRENHANCNFSM4SQ7FJGQ&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834465853%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=sTSWFPylr3n2PvjF3fufbsu0KWTny2gfUNCpFPsMNw8%3D&reserved=0>.

c

**Comment by jessegmeyerlab** [comment]
Also, I like that you added the math 
image

I'm not an expert in linear models so maybe you understand this better, but are the groups X1 and X2 the condition groups or the peptide groups? If the latter then please change "x1= peptide<sub>i</sub>" and  x2 = "all other peptides". 

If X1 and X2 are the groups then there would X3 when we have three groups?

**Comment by jessegmeyerlab** [comment]
I tested the steps in the vignette and it all works! If you can fix those two minor vingette parts I will merge this!

**Comment by demar01** [comment]
> I tested the steps in the vignette and it all works! If you can fix those two minor vingette parts I will merge this!

Awesome! I just 

> Can you see in the image of the vingette? control_name="t<-..." when it should be control_name="cntrl"
> 
> > Great, we could write in the documentation: PeCorA has been tested with R version 3.6.3. The following packages/version are required: reshape_0.8.8 standardize_0.2.1 ggplot2_3.3.2 what do you mean error in the control name? Maria Dermit Postdoctoral Research Assistant | Centre for Molecular Oncology Barts Cancer Institute - a Cancer Research UK Centre of Excellence Queen Mary, University of London John Vane Science Centre, Charterhouse Square, London EC1M 6BQ Email: [maria.dermit@qmul.ac.uk](mailto:maria.dermit@qmul.ac.uk)
> > […](#)
> > ________________________________ From: jessegmeyerlab [notifications@github.com](mailto:notifications@github.com) Sent: Monday, October 19, 2020 3:29 PM To: jessegmeyerlab/PeCorA [PeCorA@noreply.github.com](mailto:PeCorA@noreply.github.com) Cc: Maria Dermit [maria.dermit@qmul.ac.uk](mailto:maria.dermit@qmul.ac.uk); Author [author@noreply.github.com](mailto:author@noreply.github.com) Subject: Re: [jessegmeyerlab/PeCorA] First pull request (#2) OK got it to install now. Might have partially been issue with R version so I updated to 4.0.3 and ran it in the base R gui and it installed going through the vingette and found this error in the control name [image]https://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F65188012%2F96464523-c75a6700-11dc-11eb-8068-3789af3f2281.png&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834445860%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=yiZUqW1qHuXIA2SAfk37ysTi3WP6WiLyiA1qtJtVRwA%3D&reserved=0 — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHubhttps://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjessegmeyerlab%2FPeCorA%2Fpull%2F2%23issuecomment-712200875&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834455862%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=R8lVmTEzy1o4UUlhF4Cuc0qV2e4yaDKJDF38sMWib0o%3D&reserved=0, or unsubscribehttps://eur01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABGGWHPORNMDIAEZT346ABLSLRENHANCNFSM4SQ7FJGQ&data=04%7C01%7C%7Cb630449010714d0998ae08d8743b6a92%7C569df091b01340e386eebd9cb9e25814%7C0%7C0%7C637387145834465853%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=sTSWFPylr3n2PvjF3fufbsu0KWTny2gfUNCpFPsMNw8%3D&reserved=0.
> 
> c

Ups! I missed that. It is fixed on the vignette now.

**Comment by demar01** [comment]
> Also, I like that you added the math
> image

> I'm not an expert in linear models so maybe you understand this better, but are the groups X1 and X2 the condition groups or the peptide groups? If the latter then please change "x1= peptidei" and x2 = "all other peptides".
> 
> If X1 and X2 are the groups then there would X3 when we have three groups?

I think it is correct how it is  like that. It is the simplify version how we wrote it on Fig2C (image below) 

image

and the exact complete formula ( for an example peptide)  from
lm(subtmpdf$ms1adj ~ subtmpdf$Condition*subtmpdf$allothers) is:
α+ β1(subtmpdf$ConditionEtOH)+ β2(subtmpdf$ConditionLPS)+ β3(subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ β4(subtmpdf$ConditionEtOH× subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ β5(subtmpdf$ConditionLPS× subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ϵ

So there will be as many B as combinations are

**Comment by demar01** [comment]
> Also, I like that you added the math
> image
> 
> I'm not an expert in linear models so maybe you understand this better, but are the groups X1 and X2 the condition groups or the peptide groups? If the latter then please change "x1= peptidei" and x2 = "all other peptides".
> 
> If X1 and X2 are the groups then there would X3 when we have three groups?

> Also, I like that you added the math
> image

> I'm not an expert in linear models so maybe you understand this better, but are the groups X1 and X2 the condition groups or the peptide groups? If the latter then please change "x1= peptidei" and x2 = "all other peptides".
> 
> If X1 and X2 are the groups then there would X3 when we have three groups?

I think it is correct how it is  like that. It is the simplify version how we wrote it on Fig2C (image below) 

image

and the exact complete formula ( for an example peptide)  from
lm(subtmpdf$ms1adj ~ subtmpdf$Condition*subtmpdf$allothers) is:
α+ β1(subtmpdf$ConditionEtOH)+ β2(subtmpdf$ConditionLPS)+ β3(subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ β4(subtmpdf$ConditionEtOH× subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ β5(subtmpdf$ConditionLPS× subtmpdf$allothersC[+57]QGTFSPEDNSIK_all)+ϵ

So there will be as many B as combinations are

**Comment by demar01** [comment]
> I tested the steps in the vignette and it all works! If you can fix those two minor vingette parts I will merge this!

Awesome! Lemme know what you think about the formula, if you think we  need to change it in the vignette.

**Comment by demar01** [comment]
Awesome! Are we thinking of submitting the package to CRAN?

I just built a cute hexticker for the package 🙂

I will send later a new pull request to include it.



Maria Dermit
Postdoctoral Research Assistant | Centre for Molecular Oncology

Barts Cancer Institute - a Cancer Research UK Centre of Excellence
Queen Mary, University of London
John Vane Science Centre, Charterhouse Square, London EC1M 6BQ
Email: maria.dermit@qmul.ac.uk
________________________________

**Comment by demar01** [comment]
Hi Maria,

That is adorable I love it!

Yes we should submit to CRAN. Do you know how that works in case we need to update it in the future.

Best,
Jesse

**Pull request merged**